tests/mock_network/secure_message_delivery.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{create_connected_nodes_until_split, poll_all};
+use itertools::Itertools;
+use routing::{
+    mock::Network, Authority, BlsPublicKeySet, BlsPublicKeyShare, FullId, HopMessage, Message,
+    MessageContent, RoutingMessage, SectionInfo, SignedRoutingMessage,
+};
+use std::collections::BTreeSet;
+
+#[test]
+#[should_panic]
+fn message_with_invalid_signature() {
+    let min_section_size = 3;
+    let network = Network::new(min_section_size, None);
+    let mut nodes = create_connected_nodes_until_split(&network, vec![1, 1], false);
+
+    let ori_info = unwrap!(nodes[0].inner.chain()).our_info().clone();
+    let ori_prefix = *ori_info.prefix();
+
+    let mut members = ori_info.members().clone();
+    let pub_id = *FullId::new().public_id();
```

Comment:
Maybe you could try saving the full ID here, then adding it's signature share to the message, too, before combining the signatures? (if there are still problems with the test)